Core/clim-core/commands/processor.lisp

```diff
@@ -361,14 +361,31 @@
 
 (define-presentation-method accept
     ((type command) stream (view textual-view) &key)
-  (if-let ((command (funcall *command-parser* command-table stream)))
-    (values (if (partial-command-p command)
-                (funcall *partial-command-parser*
-                         command-table stream command
-                         (position *unsupplied-argument-marker* command))
-                command)
-            type)
-    (simple-parse-error "Empty command")))
+  (let ((command-table (find-command-table command-table))
+        (start-position (and (getf options :echo t)
+                             (input-editing-stream-p stream)
+                             (stream-scan-pointer stream))))
+    (flet ((maybe-replace-input (input)
+             (when start-position
+               ;; Rescan is nil in case the clicked on thing returned
+               ;; something unparseable. -- heshrobe 2020-05-15
+               (presentation-replace-input stream input type view
+                                           :buffer-start start-position
+                                           :rescan nil))
+             input)
+           (handle-command (command)
+             (if (partial-command-p command)
+                 (funcall *partial-command-parser*
+                          command-table stream command
+                          (position *unsupplied-argument-marker* command))
+                 command)))
+      (with-input-context (type)
+          (object type)
+          (if-let ((command (funcall *command-parser* command-table stream)))
+            (values (maybe-replace-input (handle-command command)) type)
+            (simple-parse-error "Empty command"))
+        (command
+         (values (maybe-replace-input (handle-command object)) type))))))
```

Comment:
Can we use the `accept.smoke` test by feeding a suitable event to the fake extended input stream?

Reply:
do you mean something like that?


```
(test accept.smoke
  (mapc
   (lambda (case)
     (destructuring-bind (input accept-type expected-value expected-type) case
       (flet ((test-stream (stream)
                (multiple-value-bind (value type)
                    (accept accept-type :stream stream :prompt nil)
                  (is (equal expected-value value)
                      "~@<ACCEPTing ~S from a ~S with type ~S resulted ~
                       in value ~S, not ~S~@:>"
                      input (type-of stream) accept-type value expected-value)
                  (is (equal expected-value value)
                      "~@<ACCEPTing ~S from a ~S with type ~S resulted ~
                       in type ~S, not ~S~@:>"
                      input (type-of stream) accept-type type expected-type)
                  (is (not (null type))
                      "~@<ACCEPTing ~S from a ~S with type ~S resulted ~
                       in type ~s, which is not a valid."))))
         ;; Test with extended input stream.
         (let ((stream (make-instance 'test-stream)))
           (map nil (alexandria:curry #'climi::stream-append-gesture stream)
                input)
           (climi::stream-append-gesture stream #\Return)
           (test-stream stream))
         ;; Test with string stream.
         (let* ((string (format nil "~A~C" input #\Return))
                (stream (make-string-input-stream string)))
           (test-stream stream)))))
   '(("1"       integer            1         integer)
     ("1,2,3,4" (sequence integer) (1 2 3 4) (sequence integer)))))
```

or do you mean, that we should define a command table and a command, then try to parse it? that won't work for partial command parsers (and that's where we were bitten), because they invoke accepting-values dialog.

Reply:
If I understand the changed code correctly, we would have to add cases that replace feeding `"1,2,3,4"` to the stream by feeding a simulated presentation-click event to it.

Check the second return value can be done in all cases it seems.

Reply:
Too difficult with the current test infrastructure.